pool/position_modify.gno

```diff
@@ -8,9 +8,24 @@ import (
 	plp "gno.land/p/gnoswap/pool"
 )
 
-// modifyPosition updates a position in the pool and calculates the amount of tokens to be added or removed.
-// Returns positionInfo, amount0, amount1
+// modifyPosition updates a position in the pool and calculates the amount of tokens
+// needed (for minting) or returned (for burning). The calculation depends on the current
+// price (tick) relative to the position's price range.
+//
+// The function handles three cases:
+//  1. Current price below range (tick < tickLower): only token0 is used/returned
+//  2. Current price in range (tickLower <= tick < tickUpper): both tokens are used/returned
+//  3. Current price above range (tick >= tickUpper): only token1 is used/returned
+//
+// Parameters:
+//   - params: ModifyPositionParams containing owner, tickLower, tickUpper, and liquidityDelta
+//
+// Returns:
+//   - PositionInfo: updated position information
+//   - *u256.Uint: amount of token0 needed/returned
+//   - *u256.Uint: amount of token1 needed/returned
 func (pool *Pool) modifyPosition(params ModifyPositionParams) (PositionInfo, *u256.Uint, *u256.Uint) {
```

Comment:
Curious, what's the difference between `func (pool *Pool)` here and `func (p *Pool)` in `tick.gno`? I see both are being used. If it doesn't have any meaning, then we can unify it?